### PR TITLE
7189422: [macosx] Submenu's arrow have a wrong position

### PR DESCRIPTION
**Pull request by prsadhuk**
Issue is Arrow in submenu with empty title have a wrong position in Aqua L&F as can be seen

image

which is because the text being null/empty, `labelR` rectangle width/height is 0 so arrowIcon y coordinate becomes -ve as per the calculation in layoutMenuItem(). Although it seems logical to me to not show the arrow if submenu text is null, but
whereas for other L&F, it is shown as this 
for Metal
image

for Nimbus
image

so the fix is made in Aqua L&F to show as other L&F as
image

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-7189422](https://bugs.openjdk.org/browse/JDK-7189422): [macosx] Submenu's arrow have a wrong position

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9769/head:pull/9769` \
`$ git checkout pull/9769`

Update a local copy of the PR: \
`$ git checkout pull/9769` \
`$ git pull https://git.openjdk.org/jdk pull/9769/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9769`

View PR using the GUI difftool: \
`$ git pr show -t 9769`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9769.diff">https://git.openjdk.org/jdk/pull/9769.diff</a>

</details>

### CODE REVIEW

File: src/java.desktop/macosx/classes/com/apple/laf/AquaMenuPainter.java [hunk 1]
```diff
@@ -494,13 +494,15 @@ private String layoutMenuItem(final JMenuItem menuItem, final FontMetrics fm, fi
         if (!isTopLevelMenu) {
             //    if ( GetSysDirection() < 0 ) hierRect.right = hierRect.left + w + 4;
             //    else hierRect.left = hierRect.right - w - 4;
```
Comment (AJ1032480) [inline src/java.desktop/macosx/classes/com/apple/laf/AquaMenuPainter.java:496]: delete unused code

**Review by AJ1032480 — CHANGES_REQUESTED** [review]

File: src/java.desktop/macosx/classes/com/apple/laf/AquaMenuPainter.java [hunk 2]
```diff
@@ -494,13 +494,15 @@ private String layoutMenuItem(final JMenuItem menuItem, final FontMetrics fm, fi
...
-            arrowIconR.x = (viewR.width - arrowIconR.width) + 1;
-            arrowIconR.y = viewR.y + (labelR.height / 2) - (arrowIconR.height / 2) + 1;
 
-            checkIconR.y = viewR.y + (labelR.height / 2) - (checkIconR.height / 2);
+            arrowIconR.x = Math.abs((viewR.width - arrowIconR.width) + 1);
```
Comment (prrace) [inline src/java.desktop/macosx/classes/com/apple/laf/AquaMenuPainter.java:498]: May I take it that you verified this doesn't have any adverse effects on submenus with text :?

Comment (prsadhuk) [reply src/java.desktop/macosx/classes/com/apple/laf/AquaMenuPainter.java:498]: Yes, with fix it is
Screenshot 2022-08-08 at 9 46 02 AM

whereas without fix it is
Screenshot 2022-08-08 at 9 47 57 AM

**Review by prrace — APPROVED** [review]

**Review by DamonGuy — APPROVED** [review]
Applied your changes and ran the test. The fix works as described.

**Pull request closed without merging**